### PR TITLE
Adding asdot notation to ASN views

### LINKED ISSUE #8293: support for asdot notation for 4 Byte ASN
### NetBox version

v3.1.5

### Feature type

Change to existing functionality

### Proposed functionality

In the netbox url/ipam/asns/ and  netbox ur>/ipam/asns/#/ view the AS Number should be displayed in asplain and asdot notation. 
The db model does not have to be changed as we can still only store the asplain value in the db.

The actual conversion can be done in the web browser with some simple javascript like this:

```
 function conv_asplain2asdot() {
        asplain = document.AScalc.asplain.value;
        var asdot1 = (Math.floor(asplain/65536));
        var asdot2 = asplain%65536;
        document.AScalc.asdot1.value = asdot1;
        document.AScalc.asdot2.value = asdot2;
};

function conv_asdot2asplain() {
        var asdot1 = document.AScalc.asdot1.value;
        var asdot2 = document.AScalc.asdot2.value;
        asplain = ((parseInt(asdot1) * 65536) + parseInt(asdot2));
        document.AScalc.asplain.value = asplain;
};
```

### Use case

4 Byte AS number get more popular these day and the asdot notation seems to be much easier to work with. 

### Database changes

not required, conversion can be done in javascript at runtime as mentioned above.

### External dependencies

none

### PR DESCRIPTION
**Pull request by jasonyates**
### Fixes: #8293 
Adds custom property to asn model to compute asdot notation if required.
Updates asn view to show asdot notation if one exists in the format xxxxx (yyy.yyy)
Adds a custom column renderer to asn table to display asdot notation if one exists

### CODE REVIEW

**Comment by jasonyates** [comment]
@jeremystretch Soliciting some feedback on this.

Should the page title also be updated to include the asdot notation or not?

E.g.
Screenshot 2022-01-12 at 14 06 36

**Comment by jeremystretch** [comment]
Sure? No strong opinion here.

File: netbox/ipam/models/ip.py [hunk 1]
```diff
@@ -125,11 +125,22 @@ class Meta:
         verbose_name_plural = 'ASNs'
 
     def __str__(self):
-        return f'AS{self.asn}'
+        if self.asn > 65535:
+            return 'AS{} ({}.{})'.format(self.asn, self.asn // 65536, self.asn % 65536)
```
Comment (jeremystretch) [inline netbox/ipam/models/ip.py:129]: This should probably return `self.asdot_notation`.

File: netbox/ipam/models/ip.py [hunk 2]
```diff
@@ -125,11 +125,22 @@ class Meta:
...
+        else:
+            return f'AS{self.asn}'
 
     def get_absolute_url(self):
         return reverse('ipam:asn', args=[self.pk])
 
+    @property
+    def asdot_notation(self):
```
Comment (jeremystretch) [inline netbox/ipam/models/ip.py:137]: Maybe rename this to `asn_with_asdot` or something like that, and return either the ASN with ASDOT notiation or just the ASN. Then you can reference it directly from within the ASNTable and object template.

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Thanks for this! Just had a couple suggestions.

File: netbox/ipam/models/ip.py (hunk 1 above)
Comment (DanSheps) [reply netbox/ipam/models/ip.py:129]: I think we should also stick with fstrings if we can IMO

Comment (jasonyates) [reply netbox/ipam/models/ip.py]: Updated to use fstring and to return `asn_with_asdot` per below suggestion. Resulting code is much cleaner.

File: netbox/ipam/models/ip.py (hunk 2 above)
Comment (jasonyates) [reply netbox/ipam/models/ip.py]: Great suggestion. That works much cleaner. Updated ASNTable & object template.

**Review by jeremystretch — APPROVED** [review]

**Comment by jeremystretch** [comment]
Excellent, thanks @jasonyates!

**Pull request merged**